nostr/relay.py

```diff
@@ -54,42 +51,55 @@ def __post_init__(self):
             on_error=self._on_error,
             on_close=self._on_close
         )
+        self._connection_thread: Thread = None
 
-    def connect(self):
-        self.ws.run_forever(
-            sslopt=self.ssl_options,
-            http_proxy_host=self.proxy_config.host if self.proxy_config is not None else None, 
-            http_proxy_port=self.proxy_config.port if self.proxy_config is not None else None,
-            proxy_type=self.proxy_config.type if self.proxy_config is not None else None,
-        )
+    def connect(self, is_reconnect=False):
+        if not self.is_connected():
+            with self.lock:
+                self._connection_thread = Thread(
+                    target=self.ws.run_forever,
+                    kwargs={
+                        "sslopt": self.ssl_options,
+                        "http_proxy_host": self.proxy_config.host if self.proxy_config is not None else None,
+                        "http_proxy_port": self.proxy_config.port if self.proxy_config is not None else None,
+                        "proxy_type": self.proxy_config.type if self.proxy_config is not None else None
+                    },
+                    name=f"{self.url}-connection"
+                )
+                self._connection_thread.start()
+
+            if not is_reconnect:
+                Thread(
+                    target=self.outgoing_messages_worker,
+                    name=f"{self.url}-outgoing-messages-worker",
+                    daemon=True
+                ).start()
 
+            time.sleep(1)
+        
     def close(self):
-        self.ws.close()
+        if self.is_connected():
+            self.ws.close()
 
-    def check_reconnect(self):
-        try:
-            self.close()
-        except:
-            pass
-        self.connected = False
-        if self.reconnect:
-            time.sleep(1)
-            self.connect()
+    def is_connected(self) -> bool:
+        with self.lock:
+            if self._connection_thread is None or not self._connection_thread.is_alive():
+                return False
+            else:
+                return True
 
     def publish(self, message: str):
-        self.queue.put(message)
+        self.outgoing_messages.put(message)
         
-    def queue_worker(self):
+    def outgoing_messages_worker(self):
         while True:
-            if self.connected:
-                message = self.queue.get()
+            if self.is_connected():
+                message = self.outgoing_messages.get()
```

Comment:
Add comment here that `Queue.get()` blocks and waits by default. Was confusing to read this without that prior knowledge.

Also obv verify that the blocking still allows this thread to terminate when the main thread exits.

Reply:
I don't think it's necessary since that is the normal way any Queue would work.